tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -88,14 +90,15 @@ var _ = Describe("odo devfile push command tests", func() {
 			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
 
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+			cmpName := helper.RandString(6)
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, "--namespace", namespace, cmpName)
```

Comment:
This command shouldn't pass based on the design https://github.com/openshift/odo/issues/2789 we discussed before.

```
If devfile.yaml is present
Case 1: odo create <component type> -> Throw errors and tell user to delete the existing devfile.yaml then run odo create again
```
If you want to specify the component type, you have to delete the existing devfile.yaml as component type is used for downloading the devfile.yaml. Or you can run odo create without component type if you want to leave the devfile.yaml

Reply:
same as: https://github.com/openshift/odo/pull/2762#discussion_r401864713